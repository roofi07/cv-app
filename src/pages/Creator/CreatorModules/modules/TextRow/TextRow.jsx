import React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import PropTypes from 'prop-types'
import './TextRow.scss'

export class TextRow extends React.Component {
  static propTypes = {
    data: ImmutablePropTypes.map,
    onUpdate: PropTypes.func,
  }

  constructor() {
    super()

    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(e) {
    const { onUpdate } = this.props
    let { data } = this.props
    const { id, value } = e.target

    data = data.set(id, value)
    onUpdate(data)
  }

  render() {
    const { data } = this.props
    return (
      <div className="textrow">
        <div className="textrow__key">
          <input id="key" onChange={this.handleChange} placeholder="Enter descprition..." value={data.get('key', '')} />
        </div>
        <div className="textrow__value">
          <input id="value" onChange={this.handleChange} placeholder="Enter content..." value={data.get('value', '')} />
        </div>
      </div>
    )
  }
}

export default TextRow
