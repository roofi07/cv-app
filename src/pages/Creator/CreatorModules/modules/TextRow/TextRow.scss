.textrow {
  display: flex;

  &__key,
  &__value {
    padding: 4px;

    input {
      display: block;
      width: 100%;
      padding: 6px;
      border: 0;
      border: 1px solid transparent;

      &:focus {
        outline: 0
      }

      &:hover,
      &:focus {
        border-color: #dedede;        
      }
    }
  }

  &__key {
    width: 200px;
    text-align: right;

    input {
      text-align: right;
    }
  }

  &__value {
    flex: 1;
  }
}
