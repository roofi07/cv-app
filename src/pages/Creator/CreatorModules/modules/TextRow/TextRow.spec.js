import React from 'react'
import { shallow } from 'enzyme'
import Immutable from 'immutable'
import { TextRow } from './TextRow'

const data = {
  key: 'Test',
  value: 'test',
}

const initPage = overrides => {
  const mockMethods = {
    onUpdate: jest.fn(),
  }
  const wrapper = shallow(<TextRow
    data={Immutable.fromJS(data)}
    {...mockMethods}
    {...overrides}
  />)
  return { mockMethods, wrapper }
}

describe('TextRow', () => {
  it('should render as expected', () => {
    const { wrapper } = initPage()
    expect(wrapper).toMatchSnapshot()
  })
})
