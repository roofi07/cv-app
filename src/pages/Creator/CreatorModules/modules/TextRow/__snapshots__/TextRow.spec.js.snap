// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextRow should render as expected 1`] = `
<div
  className="textrow"
>
  <div
    className="textrow__key"
  >
    <input
      id="key"
      onChange={[Function]}
      placeholder="Enter descprition..."
      value="Test"
    />
  </div>
  <div
    className="textrow__value"
  >
    <input
      id="value"
      onChange={[Function]}
      placeholder="Enter content..."
      value="test"
    />
  </div>
</div>
`;
